src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -400,17 +411,20 @@ private void Unregister()
                 _registeredEvents = SocketAsyncEvents.None;
                 _handle.Free();
             }
+            else
+            {
+                Debug.Fail(string.Format("Unregister failed: {0}", errorCode));
+            }
         }
 
-        public void Close()
+        private void ProcessClose()
```

Comment:
Nit: Consider a different name, like CloseCore, CloseInner, ClosePrivate, something like that.  Took me a minute to realize that this was about processing the close, not about closing the process ;)